.github/workflows/ci.yaml

```diff
@@ -0,0 +1,116 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  test:
+    if: "! contains(toJSON(github.event.commits.*.message), '[skip-ci]')"
+    runs-on: ubuntu-latest
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    steps:
+
+      - name: Git Checkout
+        id: git_checkout
+        uses: actions/checkout@v2
+
+      - name: Setup Python
+        id: setup_python
+        uses: actions/setup-python@v1
+        with:
+          python-version: '3.7'
+
+      - name: Setup Test Environment
+        id: setup_test_environment
+        run: |
+          python -m pip install pyyaml cwltest cwlref-runner
+          short_sha=${GITHUB_SHA::8}
+          echo "::set-env name=SHORT_SHA::$short_sha"
+
+      - name: Build Docker Image For testing
+        id: docker_build
+        uses: docker/build-push-action@v1
+        with:
+          repository: tessthyer/${{ github.event.repository.name }}
+          push: false # don't push before testing
+          tags: ${{ env.SHORT_SHA }}
+
+      - name: Edit CWL file tool version
+        id: edit_cwl
+        run: |
+          tools_dir=cwl
+          new_version=${{ env.SHORT_SHA }}
+          python utils/bump_cwl_version.py $tools_dir $new_version
+
+      - name: Test CWL
+        id: test_cwl
+        run: |
+          cwltest --test test-descriptions.yaml --tool cwl-runner
+
+  build:
+    if: "! contains(toJSON(github.event.commits.*.message), '[skip-ci]')"
+    needs: test
+    runs-on: ubuntu-latest
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    steps:
+
+      - name: Git Checkout
+        id: git_checkout
+        uses: actions/checkout@v2
+
+      - name: Setup Python
+        id: setup_python
+        uses: actions/setup-python@v1
+        with:
+          python-version: '3.7'
+
+      - name: Setup Build Environment
+        id: setup_build_environment
+        run: |
+          python -m pip install pyyaml
+
+      - name: Calculate Next Autotagging Version
+        id: calc_version
+        uses: mathieudutour/github-tag-action@v4.2
```

Comment:
Cool! Does this mean you're committing (pun intended) to using commit messages to control version bumping?

Reply:
No. The only special commit message content remains `[skip-ci]`. The major change  is that I wanted to use git tagging everywhere rather than version files.